### DISCUSSION TITLE
Steam Deck: Help detecting games after Moraroy's NonSteamLaunchers

### DISCUSSION THREAD

**Discussion by JpTiger** [PhilipK/BoilR#359] [Q&A]
Hi there, I'm glad discussions are enabled here since I'm pretty sure the problem is probably me, not BoilR. I always feels a little presumptuous writing up problems I have as bug reports in Issues.

I've installed nonsteam launchers using https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck, which also includes BoilR packed into it. I used the script to install Epic, GOG, Itch, Ubisoft Connect, Battle.net, and Origin. A couple of those I haven't bothered installing any games for since I've heard they're so much trouble (looking at you, Origin). 

But I haven't gotten BoilR to successfully work with any of them so far. Of the ones I've tried after installing games, GoG and Itch.io games aren't detected even after specifying their install folders, half a dozen random Ubisoft games that aren't installed are detected and the ones I have installed aren't, and the Epic games are detected but then the shortcuts don't have any art and don't work.

I tried downloading the latest release from here to make sure it's not a problem with what's packaged in BoilR, but I get the same results (and saved preferences, so something is working that should be at least). Is this just a collection of known issues in SteamDeck/Linux, or am I doing something wrong? Any tips on how to fix?

Thank you!

**Comment by PhilipK** [comment]
So I don't quite know how BoilR does with picking up games installed with other products.

I can tell you what I did when I tested and developed it for myself and that was to download the exe for an installer, manually add it as a game in steam, launch it with Proton, install the installer, then find the compat folder that match the installed launchers and launch them directly again with steam , but with the compat folder environment variable set to use the launchers compat folder instead of making a new one.

STEAM_COMPAT_DATA_PATH
Is the name of the folder as far as I remember.

If you do that it should be able to pick up the games

**Reply by JpTiger** [reply]
Just had an exchange with the maker of the script on reddit where I pointed him to this github discussion and asked if that was how his script worked. He said yes ([here](https://www.reddit.com/r/SteamDeck/comments/15f0exh/nonsteamlaunchers_v298_update_adds_compatability/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=1), comment thread starting with Tibarn93). Given that, any ideas how to troubleshoot?

**Reply by moraroy** [reply]
@PhilipK @JpTiger BoilR should pick up the games using my script. However, BoilR is unable to pick up some of the games from some of the launchers for some time now. I myself have used both the Linux Version and Windows Version on my deck in order to satisfy the 5 launchers to create shortcuts for my games. I personally would throw the Windows version into the prefix folder and run it from there using proton in game mode. My Script, NonSteamLaunchers does nothing different than what @PhilipK has pointed out on how he would install the launcher.

**Comment by moraroy** [comment]
@PhilipK hey Phillip I kind of found a weird work around for the steam deck just wanted to see what you thought about it... so I was able to install playnite on my deck inside of a proton prefix. Since playnite doesn't have linux support alot of the functionality doesn't work. However, the integrations still do. So once I added my integrations in playnite. I noticed in the (Windows version) of Boilr all of my games showed up under "Playnite". This is good because all my games were there from every launcher. The problem is since this is the Windows version of Boilr the compat data command isn't in the launch options. Was wondering if it's possible to give this functionality of being able to detect Playnite on the "Linux" version of Boilr so it can just detect playnite in Proton. That way the linux users and us steam deck folks could use BoilR(Linux) while piggybacking off of Playnite(Proton). BTW I installed playnite as a nonsteam game and ran it in safe mode. So there is no functionality with playnite really, just taking advantage of the integrations without really using the program.
